src/widget/paintable.cpp

```diff
@@ -53,7 +53,12 @@ Paintable::Paintable(const PixmapSource& source, DrawMode mode, double scaleFact
         : m_drawMode(mode),
           m_source(source) {
     if (!source.isSVG()) {
-        m_pPixmap.reset(WPixmapStore::getPixmapNoCache(source.getPath(), scaleFactor));
+            auto pPixmap = WPixmapStore::getPixmapNoCache(source.getPath(), scaleFactor);
+            if (!pPixmap) {
+                qWarning() << "Failed to load pixmap from path:" << source.getPath();
+                return;
+            }
+            m_pPixmap.reset(pPixmap.release());
```

Comment:
no need to overcomplicate things. `unique_ptr` does the right thing (potentially even faster than the manual reset+release).
```suggestion
            m_pPixmap = pPixmap;
```

Reply:
whoops. I didn't notice that m_pPixmap is Qts unique ptr and  `pPixmap` is a `std::unique_ptr`... I'm sorry for the bogus suggestion then. Feel free to revert/drop this suggestion or make them both `std::unique_ptr`, whatever you prefer.